core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -73,7 +73,7 @@ class Replica(val brokerId: Int,
    * high frequency.
    */
   def updateLogReadResult(logReadResult: LogReadResult) {
-    if (logReadResult.info.fetchOffsetMetadata.messageOffset >= logReadResult.leaderLogEndOffset)
+    if (logReadResult.info.fetchOffsetMetadata.messageOffset >= logReadResult.leaderLogEndOffset && logReadResult.leaderLogEndOffset >= 0)
```

Comment:
I guess this is not strictly needed since `_lastCaughtUpTimeMs` cannot become negative?

Reply:
In `Partition.makeLeader()`, there is `updateLogReadResult(LogReadResult.UnknownLogReadResult)`, where the `leaderLogEndOffset` is negative.

Reply:
Yeah you are right. I couldn't remember why this change was made. I have removed this change.